src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -376,10 +380,49 @@ public bool TryGetVariableType(string name, out FormulaType type)
 
             return false;
         }
-
+        
         public void UpdateSupportedFunctions(SymbolTable s)
         {
             SupportedFunctions = s;
         }
+
+        /// <summary>
+        /// Add a set of user-defined formulas and functions to the engine.
+        /// </summary>
+        /// <param name="script">Script containing user defined functions and/or named formulas.</param>
+        /// <param name="parseCulture">Locale to parse user defined script.</param>
+        /// <param name="symbolTable">Extra SymbolTable if needed.</param>
+        /// <param name="onUpdate">Function to be called when update is triggered.</param>
+        public void AddUserDefinitions(string script, CultureInfo parseCulture = null, ReadOnlySymbolTable symbolTable = null, Action<string, FormulaValue> onUpdate = null)

```

Comment:
>AddUserDefinitions [](http://example.com/codeflow?start=20&length=18)

can we move to engine base class?
For parse-only scenarios. And so we can work with different backends.